src/main/java/com/google/firebase/projectmanagement/FirebaseProjectManagementServiceImpl.java

```diff
@@ -0,0 +1,909 @@
+      }
+    };
+  }
+
+  /**
+   * An {@link ApiAsyncFunction} that transforms a Long Running Operation name to an {@link IosApp}
+   * or an {@link AndroidApp} instance by repeatedly polling the server (with exponential backoff)
+   * until the App is created successfully, or until the number of poll attempts exceeds the maximum
+   * allowed.
+   */
+  private class WaitOperationFunction implements ApiAsyncFunction<String, String> {
+    private final String projectId;
+
+    private WaitOperationFunction(String projectId) {
+      this.projectId = projectId;
+    }
+
+    /**
+     * Returns an {@link ApiFuture} that will eventually contain the App ID of the new created App,
+     * or an exception if an error occurred during polling.
+     */
+    @Override
+    public ApiFuture<String> apply(String operationName) throws FirebaseProjectManagementException {
+      SettableApiFuture<String> settableFuture = SettableApiFuture.<String>create();
+      scheduledExecutor.schedule(
+          new WaitOperationRunnable(
+              /* numberOfPreviousPolls= */ 0,
+              operationName,
+              projectId,
+              settableFuture),
+          /* delay= */ 0L,
+          TimeUnit.MILLISECONDS);
+      return settableFuture;
+    }
+  }
+
+  /**
+   * A poller that repeatedly polls a Long Running Operation (with exponential backoff) until its
+   * status is "done", or until the number of polling attempts exceeds the maximum allowed.
+   */
+  private class WaitOperationRunnable implements Runnable {
+    private final int numberOfPreviousPolls;
+    private final String operationName;
+    private final String projectId;
+    private final SettableApiFuture<String> settableFuture;
+
+    private WaitOperationRunnable(
+        int numberOfPreviousPolls,
+        String operationName,
+        String projectId,
+        SettableApiFuture<String> settableFuture) {
+      this.numberOfPreviousPolls = numberOfPreviousPolls;
+      this.operationName = operationName;
+      this.projectId = projectId;
+      this.settableFuture = settableFuture;
+    }
+
+    @Override
+    public void run() {
+      String url = String.format("%s/v1/%s", FIREBASE_PROJECT_MANAGEMENT_URL, operationName);
+      OperationResponse operationResponseInstance;
+      operationResponseInstance = new OperationResponse();
+      try {
+        makeGetRequest(url, operationResponseInstance, projectId, "Project ID");
+      } catch (FirebaseProjectManagementException e) {
+        settableFuture.setException(e);
+        return;
+      }
+      if (!operationResponseInstance.done) {
+        if (numberOfPreviousPolls + 1 >= MAXIMUM_POLLING_ATTEMPTS) {
+          settableFuture.setException(createFirebaseProjectManagementException(
+              projectId,
+              "Project ID",
+              "Unable to create App: deadline exceeded.",
+              /* cause= */ null));
+        } else {
+          long delay = (long) (
```

Comment:
The exponential growth of the delay should be capped at some maximum. I think the Node.js implementation of this API uses a max of 10 seconds.